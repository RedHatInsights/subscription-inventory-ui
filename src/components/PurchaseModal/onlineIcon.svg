<svg id="f2dc99c9-3f09-4f6a-892b-49321c747e9f" data-name="Icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 36 36">
  <defs>
    <style>
      .ab89ba59-1b88-4d27-ab9d-80b0c5f89f97 {
        fill: #e00;
      }
    </style>
  </defs>
  <g>
    <path class="ab89ba59-1b88-4d27-ab9d-80b0c5f89f97" d="M10,11.37a.62.62,0,0,0-.61.63.61.61,0,0,0,.61.62.62.62,0,0,0,.63-.62A.63.63,0,0,0,10,11.37Z"/>
    <path class="ab89ba59-1b88-4d27-ab9d-80b0c5f89f97" d="M8,11.37a.62.62,0,0,0-.61.63.61.61,0,0,0,.61.62A.62.62,0,0,0,8.62,12,.63.63,0,0,0,8,11.37Z"/>
    <g>
      <path class="ab89ba59-1b88-4d27-ab9d-80b0c5f89f97" d="M31,8.38H5A.62.62,0,0,0,4.38,9V27a.62.62,0,0,0,.62.62H31a.62.62,0,0,0,.62-.62V9A.62.62,0,0,0,31,8.38Zm-.62,18H5.62V9.62H30.38Z"/>
      <path class="ab89ba59-1b88-4d27-ab9d-80b0c5f89f97" d="M18,15.38H8a.62.62,0,0,0-.62.62v8a.62.62,0,0,0,.62.62H18a.62.62,0,0,0,.62-.62V16A.62.62,0,0,0,18,15.38Zm-.62,8H8.62V16.62h8.76Z"/>
      <path class="ab89ba59-1b88-4d27-ab9d-80b0c5f89f97" d="M13,12.62H28a.62.62,0,1,0,0-1.24H13a.62.62,0,0,0,0,1.24Z"/>
      <path class="ab89ba59-1b88-4d27-ab9d-80b0c5f89f97" d="M28,16.38H21a.62.62,0,0,0,0,1.24h7a.62.62,0,0,0,0-1.24Z"/>
      <path class="ab89ba59-1b88-4d27-ab9d-80b0c5f89f97" d="M28,19.38H21a.62.62,0,0,0,0,1.24h7a.62.62,0,0,0,0-1.24Z"/>
      <path class="ab89ba59-1b88-4d27-ab9d-80b0c5f89f97" d="M24,22.38H21a.62.62,0,0,0,0,1.24h3a.62.62,0,0,0,0-1.24Z"/>
    </g>
  </g>
</svg>
