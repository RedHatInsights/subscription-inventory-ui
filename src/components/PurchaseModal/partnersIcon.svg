<svg id="f05479e7-a019-44f1-8968-b41df6276f9e" data-name="Icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 36 36">
  <defs>
    <style>
      .b7a71d9a-e907-4422-b89e-c93efee7d2cc {
        fill: #e00;
      }
    </style>
  </defs>
  <g>
    <path class="b7a71d9a-e907-4422-b89e-c93efee7d2cc" d="M12.87,23.29a3.86,3.86,0,1,0-4.74,0A7.11,7.11,0,0,0,3.38,30a.61.61,0,0,0,.62.62H17a.63.63,0,0,0,.63-.62A7.13,7.13,0,0,0,12.87,23.29Zm-5-3a2.62,2.62,0,0,1,5.24,0A2.62,2.62,0,0,1,7.88,20.25ZM4.66,29.38c.84-6.94,10.84-6.93,11.68,0Z"/>
    <path class="b7a71d9a-e907-4422-b89e-c93efee7d2cc" d="M13.37,15.38a6,6,0,0,1,9.48.18.61.61,0,0,0,.66-.07c1.07-1-2.27-3-3.13-3.21a3.86,3.86,0,1,0-4.76,0c-.86.25-4.2,2.18-3.13,3.21A.62.62,0,0,0,13.37,15.38Zm2-6.13a2.62,2.62,0,0,1,5.24,0A2.62,2.62,0,0,1,15.38,9.25Z"/>
    <path class="b7a71d9a-e907-4422-b89e-c93efee7d2cc" d="M16,19.38a.62.62,0,0,0,0,1.24h4a.62.62,0,0,0,0-1.24Z"/>
    <path class="b7a71d9a-e907-4422-b89e-c93efee7d2cc" d="M27.87,23.29a3.86,3.86,0,1,0-4.74,0A7.11,7.11,0,0,0,18.38,30a.61.61,0,0,0,.62.62H32a.63.63,0,0,0,.63-.62A7.13,7.13,0,0,0,27.87,23.29Zm-5-3a2.62,2.62,0,0,1,5.24,0A2.62,2.62,0,0,1,22.88,20.25Zm-3.22,9.13c.84-6.94,10.84-6.93,11.68,0Z"/>
  </g>
</svg>
