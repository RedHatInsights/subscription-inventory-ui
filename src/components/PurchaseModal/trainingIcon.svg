<svg id="ebd7c5ab-2ab1-4013-9461-d431819ace7a" data-name="Icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 36 36">
  <defs>
    <style>
      .a54d53a2-da0e-4141-a650-cbe1d8c4cd58 {
        fill: #e00;
      }
    </style>
  </defs>
  <g>
    <path class="a54d53a2-da0e-4141-a650-cbe1d8c4cd58" d="M13.87,24.29a3.88,3.88,0,1,0-4.74,0A7.12,7.12,0,0,0,4.38,31a.62.62,0,0,0,.62.62H18a.62.62,0,0,0,.62-.62A7.12,7.12,0,0,0,13.87,24.29Zm-5-3a2.62,2.62,0,1,1,2.62,2.63A2.62,2.62,0,0,1,8.88,21.25ZM5.66,30.38a5.87,5.87,0,0,1,11.68,0Z"/>
    <path class="a54d53a2-da0e-4141-a650-cbe1d8c4cd58" d="M31,4.38H12a.62.62,0,0,0-.62.62V15a.62.62,0,0,0,1.24,0V5.62H30.38V23.38H18a.62.62,0,0,0,0,1.24H31a.62.62,0,0,0,.62-.62V5A.62.62,0,0,0,31,4.38Z"/>
  </g>
</svg>
