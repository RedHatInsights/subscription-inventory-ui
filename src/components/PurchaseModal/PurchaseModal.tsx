import React, { FunctionComponent, useState } from 'react';
import {
  Button,
  Card,
  CardBody,
  CardFooter,
  CardTitle,
  Flex,
  FlexItem,
  Grid,
  GridItem,
  Modal,
  ModalVariant,
  Title
} from '@patternfly/react-core';
import ExternalLink from '../ExternalLink';
import onlineIcon from './onlineIcon.svg';
import salesIcon from './salesIcon.svg';
import partnersIcon from './partnersIcon.svg';
import trainingIcon from './trainingIcon.svg';
import tryIcon from './tryIcon.svg';

const PurchaseModal: FunctionComponent = () => {
  const [isOpen, setIsOpen] = useState(false);

  const handleToggle = () => {
    setIsOpen(!isOpen);
  };

  interface SectionHeadingProps {
    icon: string;
    title: string;
  }

  const SectionHeading: FunctionComponent<SectionHeadingProps> = ({ icon, title }) => {
    return (
      <Flex
        alignItems={{ default: 'alignItemsFlexStart' }}
        grow={{ default: 'grow' }}
        spaceItems={{ default: 'spaceItemsXs' }}
      >
        <FlexItem style={{ height: '30px' }}>
          <img src={icon} style={{ height: '30px' }} />
        </FlexItem>
        <FlexItem>
          <Title headingLevel="h2" size="lg">
            {title}
          </Title>
        </FlexItem>
      </Flex>
    );
  };

  interface PurchaseCardProps {
    body: string;
    ctaHref: string;
    ctaText: string;
    title: string;
  }

  const PurchaseCard: FunctionComponent<PurchaseCardProps> = ({
    body,
    ctaHref,
    ctaText,
    title
  }) => {
    return (
      <Card isFullHeight>
        <CardTitle>{title}</CardTitle>
        <CardBody>{body}</CardBody>
        <CardFooter>
          <ExternalLink href={ctaHref} variant="primary">
            {ctaText}
          </ExternalLink>
        </CardFooter>
      </Card>
    );
  };

  return (
    <>
      <Button variant="primary" onClick={handleToggle}>
        Purchase subscriptions
      </Button>
      <Modal
        isOpen={isOpen}
        onClose={handleToggle}
        title="Purchase Subscriptions"
        variant={ModalVariant.medium}
      >
        <Grid hasGutter>
          <GridItem span={12}>
            <SectionHeading icon={onlineIcon} title="Online" />
          </GridItem>
          <GridItem span={6}>
            <PurchaseCard
              body="A simpler way to buy and manage enterprise software, with automated deployment to any cloud."
              ctaHref="https://marketplace.redhat.com"
              ctaText="Go to the Marketplace"
              title="Red Hat Marketplace"
            />
          </GridItem>
          <GridItem span={6}>
            <PurchaseCard
              body="Easily configure and buy Red Hat Enterprise Linux in our online store."
              ctaHref="https://www.redhat.com/store"
              ctaText="Go to the Red Hat Store"
              title="Red Hat Store"
            />
          </GridItem>
          <GridItem span={6}>
            <SectionHeading icon={salesIcon} title="Sales" />
          </GridItem>
          <GridItem span={6}>
            <SectionHeading icon={partnersIcon} title="Partners" />
          </GridItem>
          <GridItem span={6}>
            <PurchaseCard
              body="Get expert help configuring Red Hat Enterprise Linux to suit your needs."
              ctaHref="https://access.redhat.com/support/contact/sales"
              ctaText="Contact Sales"
              title="From a Red Hatter"
            />
          </GridItem>
          <GridItem span={6}>
            <PurchaseCard
              body="Find and work with an authorized Red Hat partner"
              ctaHref="https://redhat.secure.force.com/finder"
              ctaText="Search for partners"
              title="Through our partners"
            />
          </GridItem>
          <GridItem span={6}>
            <SectionHeading icon={trainingIcon} title="Training" />
          </GridItem>
          <GridItem span={6}>
            <SectionHeading icon={tryIcon} title="Try" />
          </GridItem>
          <GridItem span={6}>
            <PurchaseCard
              body="Get access to more than 1,000 instructor-led recordings, training courses, and up to 400 hours of hands-on labs - which can all be completed online at your own pace."
              ctaHref="https://www.redhat.com/training/learning-subscription"
              ctaText="Learn more"
              title="Red Hat Learning Subscription"
            />
          </GridItem>
          <GridItem span={6}>
            <PurchaseCard
              body="Our no-cost trials help you gain hands-on experience, prepare for a certification, or assess if a product is right for your organization."
              ctaHref="https://www.redhat.com/en/products/trials"
              ctaText="Go to product trial center"
              title="Red Hat product trial center"
            />
          </GridItem>
        </Grid>
      </Modal>
    </>
  );
};

export default PurchaseModal;
