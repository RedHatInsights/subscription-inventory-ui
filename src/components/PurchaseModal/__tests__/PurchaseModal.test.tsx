import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import PurchaseModal from '../PurchaseModal';

jest.mock('../onlineIcon.svg', () => 'Online Icon');
jest.mock('../salesIcon.svg', () => 'Sales Icon');
jest.mock('../partnersIcon.svg', () => 'Partners Icon');
jest.mock('../trainingIcon.svg', () => 'Training Icon');
jest.mock('../tryIcon.svg', () => 'Try Icon');

describe('PurchaseModal', () => {
  it('renders a button', () => {
    render(<PurchaseModal />);

    expect(document.body).toMatchSnapshot();
  });

  it('renders a modal when the button is pressed', async () => {
    render(<PurchaseModal />);

    fireEvent.click(screen.getByText('Purchase subscriptions'));
    await screen.findAllByText('Red Hat Marketplace');
    expect(document.body).toMatchSnapshot();
  });
});
