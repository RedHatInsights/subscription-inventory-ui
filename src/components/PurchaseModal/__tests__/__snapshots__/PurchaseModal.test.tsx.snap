// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurchaseModal renders a button 1`] = `
<body>
  <div>
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-primary"
      data-ouia-component-id="OUIA-Generated-Button-primary-1"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      type="button"
    >
      Purchase subscriptions
    </button>
  </div>
  <div />
</body>
`;

exports[`PurchaseModal renders a modal when the button is pressed 1`] = `
<body
  class="pf-c-backdrop__open"
>
  <div
    aria-hidden="true"
  >
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-primary"
      data-ouia-component-id="OUIA-Generated-Button-primary-2"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      type="button"
    >
      Purchase subscriptions
    </button>
  </div>
  <div>
    <div
      class="pf-c-backdrop"
    >
      <div
        class="pf-l-bullseye"
      >
        <div
          aria-describedby="pf-modal-part-3"
          aria-labelledby="pf-modal-part-2"
          aria-modal="true"
          class="pf-c-modal-box pf-m-md"
          data-ouia-component-id="OUIA-Generated-Modal-medium-2"
          data-ouia-component-type="PF4/ModalContent"
          data-ouia-safe="true"
          id="pf-modal-part-1"
          role="dialog"
        >
          <button
            aria-disabled="false"
            aria-label="Close"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 352 512"
              width="1em"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              />
            </svg>
          </button>
          <header
            class="pf-c-modal-box__header"
          >
            <h1
              class="pf-c-modal-box__title"
              id="pf-modal-part-2"
            >
              
              <span
                class="pf-c-modal-box__title-text"
              >
                Purchase Subscriptions
              </span>
            </h1>
          </header>
          <div
            class="pf-c-modal-box__body"
            id="pf-modal-part-3"
          >
            <div
              class="pf-l-grid pf-m-gutter"
            >
              <div
                class="pf-l-grid__item pf-m-12-col"
              >
                <div
                  class="pf-l-flex pf-m-space-items-xs pf-m-grow pf-m-align-items-flex-start"
                >
                  <div
                    class=""
                    style="height: 30px;"
                  >
                    <img
                      src="Online Icon"
                      style="height: 30px;"
                    />
                  </div>
                  <div
                    class=""
                  >
                    <h2
                      class="pf-c-title pf-m-lg"
                      data-ouia-component-id="OUIA-Generated-Title-1"
                      data-ouia-component-type="PF4/Title"
                      data-ouia-safe="true"
                    >
                      Online
                    </h2>
                  </div>
                </div>
              </div>
              <div
                class="pf-l-grid__item pf-m-6-col"
              >
                <article
                  class="pf-c-card pf-m-full-height"
                  data-ouia-component-id="OUIA-Generated-Card-1"
                  data-ouia-component-type="PF4/Card"
                  data-ouia-safe="true"
                  id=""
                >
                  <div
                    class="pf-c-card__title"
                  >
                    Red Hat Marketplace
                  </div>
                  <div
                    class="pf-c-card__body"
                  >
                    A simpler way to buy and manage enterprise software, with automated deployment to any cloud.
                  </div>
                  <div
                    class="pf-c-card__footer"
                  >
                    <a
                      aria-disabled="false"
                      class="pf-c-button pf-m-primary"
                      data-ouia-component-id="OUIA-Generated-Button-primary-3"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      href="https://marketplace.redhat.com"
                      target="_blank"
                    >
                      Go to the Marketplace
                      <span
                        class="pf-c-button__icon pf-m-end"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <path
                            d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                          />
                        </svg>
                      </span>
                    </a>
                  </div>
                </article>
              </div>
              <div
                class="pf-l-grid__item pf-m-6-col"
              >
                <article
                  class="pf-c-card pf-m-full-height"
                  data-ouia-component-id="OUIA-Generated-Card-2"
                  data-ouia-component-type="PF4/Card"
                  data-ouia-safe="true"
                  id=""
                >
                  <div
                    class="pf-c-card__title"
                  >
                    Red Hat Store
                  </div>
                  <div
                    class="pf-c-card__body"
                  >
                    Easily configure and buy Red Hat Enterprise Linux in our online store.
                  </div>
                  <div
                    class="pf-c-card__footer"
                  >
                    <a
                      aria-disabled="false"
                      class="pf-c-button pf-m-primary"
                      data-ouia-component-id="OUIA-Generated-Button-primary-4"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      href="https://www.redhat.com/store"
                      target="_blank"
                    >
                      Go to the Red Hat Store
                      <span
                        class="pf-c-button__icon pf-m-end"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <path
                            d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                          />
                        </svg>
                      </span>
                    </a>
                  </div>
                </article>
              </div>
              <div
                class="pf-l-grid__item pf-m-6-col"
              >
                <div
                  class="pf-l-flex pf-m-space-items-xs pf-m-grow pf-m-align-items-flex-start"
                >
                  <div
                    class=""
                    style="height: 30px;"
                  >
                    <img
                      src="Sales Icon"
                      style="height: 30px;"
                    />
                  </div>
                  <div
                    class=""
                  >
                    <h2
                      class="pf-c-title pf-m-lg"
                      data-ouia-component-id="OUIA-Generated-Title-2"
                      data-ouia-component-type="PF4/Title"
                      data-ouia-safe="true"
                    >
                      Sales
                    </h2>
                  </div>
                </div>
              </div>
              <div
                class="pf-l-grid__item pf-m-6-col"
              >
                <div
                  class="pf-l-flex pf-m-space-items-xs pf-m-grow pf-m-align-items-flex-start"
                >
                  <div
                    class=""
                    style="height: 30px;"
                  >
                    <img
                      src="Partners Icon"
                      style="height: 30px;"
                    />
                  </div>
                  <div
                    class=""
                  >
                    <h2
                      class="pf-c-title pf-m-lg"
                      data-ouia-component-id="OUIA-Generated-Title-3"
                      data-ouia-component-type="PF4/Title"
                      data-ouia-safe="true"
                    >
                      Partners
                    </h2>
                  </div>
                </div>
              </div>
              <div
                class="pf-l-grid__item pf-m-6-col"
              >
                <article
                  class="pf-c-card pf-m-full-height"
                  data-ouia-component-id="OUIA-Generated-Card-3"
                  data-ouia-component-type="PF4/Card"
                  data-ouia-safe="true"
                  id=""
                >
                  <div
                    class="pf-c-card__title"
                  >
                    From a Red Hatter
                  </div>
                  <div
                    class="pf-c-card__body"
                  >
                    Get expert help configuring Red Hat Enterprise Linux to suit your needs.
                  </div>
                  <div
                    class="pf-c-card__footer"
                  >
                    <a
                      aria-disabled="false"
                      class="pf-c-button pf-m-primary"
                      data-ouia-component-id="OUIA-Generated-Button-primary-5"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      href="https://access.redhat.com/support/contact/sales"
                      target="_blank"
                    >
                      Contact Sales
                      <span
                        class="pf-c-button__icon pf-m-end"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <path
                            d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                          />
                        </svg>
                      </span>
                    </a>
                  </div>
                </article>
              </div>
              <div
                class="pf-l-grid__item pf-m-6-col"
              >
                <article
                  class="pf-c-card pf-m-full-height"
                  data-ouia-component-id="OUIA-Generated-Card-4"
                  data-ouia-component-type="PF4/Card"
                  data-ouia-safe="true"
                  id=""
                >
                  <div
                    class="pf-c-card__title"
                  >
                    Through our partners
                  </div>
                  <div
                    class="pf-c-card__body"
                  >
                    Find and work with an authorized Red Hat partner
                  </div>
                  <div
                    class="pf-c-card__footer"
                  >
                    <a
                      aria-disabled="false"
                      class="pf-c-button pf-m-primary"
                      data-ouia-component-id="OUIA-Generated-Button-primary-6"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      href="https://redhat.secure.force.com/finder"
                      target="_blank"
                    >
                      Search for partners
                      <span
                        class="pf-c-button__icon pf-m-end"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <path
                            d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                          />
                        </svg>
                      </span>
                    </a>
                  </div>
                </article>
              </div>
              <div
                class="pf-l-grid__item pf-m-6-col"
              >
                <div
                  class="pf-l-flex pf-m-space-items-xs pf-m-grow pf-m-align-items-flex-start"
                >
                  <div
                    class=""
                    style="height: 30px;"
                  >
                    <img
                      src="Training Icon"
                      style="height: 30px;"
                    />
                  </div>
                  <div
                    class=""
                  >
                    <h2
                      class="pf-c-title pf-m-lg"
                      data-ouia-component-id="OUIA-Generated-Title-4"
                      data-ouia-component-type="PF4/Title"
                      data-ouia-safe="true"
                    >
                      Training
                    </h2>
                  </div>
                </div>
              </div>
              <div
                class="pf-l-grid__item pf-m-6-col"
              >
                <div
                  class="pf-l-flex pf-m-space-items-xs pf-m-grow pf-m-align-items-flex-start"
                >
                  <div
                    class=""
                    style="height: 30px;"
                  >
                    <img
                      src="Try Icon"
                      style="height: 30px;"
                    />
                  </div>
                  <div
                    class=""
                  >
                    <h2
                      class="pf-c-title pf-m-lg"
                      data-ouia-component-id="OUIA-Generated-Title-5"
                      data-ouia-component-type="PF4/Title"
                      data-ouia-safe="true"
                    >
                      Try
                    </h2>
                  </div>
                </div>
              </div>
              <div
                class="pf-l-grid__item pf-m-6-col"
              >
                <article
                  class="pf-c-card pf-m-full-height"
                  data-ouia-component-id="OUIA-Generated-Card-5"
                  data-ouia-component-type="PF4/Card"
                  data-ouia-safe="true"
                  id=""
                >
                  <div
                    class="pf-c-card__title"
                  >
                    Red Hat Learning Subscription
                  </div>
                  <div
                    class="pf-c-card__body"
                  >
                    Get access to more than 1,000 instructor-led recordings, training courses, and up to 400 hours of hands-on labs - which can all be completed online at your own pace.
                  </div>
                  <div
                    class="pf-c-card__footer"
                  >
                    <a
                      aria-disabled="false"
                      class="pf-c-button pf-m-primary"
                      data-ouia-component-id="OUIA-Generated-Button-primary-7"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      href="https://www.redhat.com/training/learning-subscription"
                      target="_blank"
                    >
                      Learn more
                      <span
                        class="pf-c-button__icon pf-m-end"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <path
                            d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                          />
                        </svg>
                      </span>
                    </a>
                  </div>
                </article>
              </div>
              <div
                class="pf-l-grid__item pf-m-6-col"
              >
                <article
                  class="pf-c-card pf-m-full-height"
                  data-ouia-component-id="OUIA-Generated-Card-6"
                  data-ouia-component-type="PF4/Card"
                  data-ouia-safe="true"
                  id=""
                >
                  <div
                    class="pf-c-card__title"
                  >
                    Red Hat product trial center
                  </div>
                  <div
                    class="pf-c-card__body"
                  >
                    Our no-cost trials help you gain hands-on experience, prepare for a certification, or assess if a product is right for your organization.
                  </div>
                  <div
                    class="pf-c-card__footer"
                  >
                    <a
                      aria-disabled="false"
                      class="pf-c-button pf-m-primary"
                      data-ouia-component-id="OUIA-Generated-Button-primary-8"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      href="https://www.redhat.com/en/products/trials"
                      target="_blank"
                    >
                      Go to product trial center
                      <span
                        class="pf-c-button__icon pf-m-end"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <path
                            d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                          />
                        </svg>
                      </span>
                    </a>
                  </div>
                </article>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
